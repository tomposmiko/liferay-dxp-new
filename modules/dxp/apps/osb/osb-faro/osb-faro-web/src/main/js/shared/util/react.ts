import {isEqual} from 'lodash';

export const getDisplayName = WrappedComponent =>
	WrappedComponent.displayName || WrappedComponent.name || 'Component';

type HasChanges = <T>(
	prev: T | object,
	next: T | object,
	...keys: string[]
) => boolean;

/**
 * Compare previous state or props object by provided keys to detect changes.
 */
export const hasChanges: HasChanges = (prev = {}, next = {}, ...keys) => {
	for (const key of keys) {
		if (key in next) {
			const newVal = next[key];

			const prevVal = prev[key];

			if (!isEqual(newVal, prevVal)) {
				return true;
			}
		}
	}

	return false;
};
