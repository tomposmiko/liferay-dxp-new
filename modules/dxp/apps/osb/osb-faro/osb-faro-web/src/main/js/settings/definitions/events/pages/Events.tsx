import BasePage from 'settings/components/BasePage';
import React from 'react';
import TabsCard from '../components/TabsCard';
import {getDefinitions} from 'shared/util/breadcrumbs';

interface IEventsProps {
	groupId: string;
}

const Events: React.FC<IEventsProps> = ({groupId}) => (
	<BasePage
		breadcrumbItems={[
			getDefinitions({groupId}),
			{active: true, label: Liferay.Language.get('events')}
		]}
		groupId={groupId}
		pageDescription={Liferay.Language.get(
			'this-is-the-data-model-of-events-sent-to-analytics-cloud'
		)}
		pageTitle={Liferay.Language.get('events')}
	>
		<TabsCard groupId={groupId} />
	</BasePage>
);

export default Events;
