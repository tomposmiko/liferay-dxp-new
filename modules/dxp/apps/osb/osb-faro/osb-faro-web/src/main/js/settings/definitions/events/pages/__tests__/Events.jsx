import Events from '../Events';
import mockStore from 'test/mock-store';
import React from 'react';
import {Provider} from 'react-redux';
import {render} from '@testing-library/react';
import {StaticRouter} from 'react-router';

jest.unmock('react-dom');

describe('Events', () => {
	it('should render', () => {
		const {container} = render(
			<Provider store={mockStore()}>
				<StaticRouter>
					<Events groupId='23' />
				</StaticRouter>
			</Provider>
		);

		expect(container).toMatchSnapshot();
	});
});
