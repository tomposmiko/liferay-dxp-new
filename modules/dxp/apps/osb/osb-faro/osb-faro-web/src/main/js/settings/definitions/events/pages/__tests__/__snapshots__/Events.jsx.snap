// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Events should render 1`] = `
<div>
  <div
    class="settings-base-page-root"
  >
    <div
      class="maintenance-alert-root"
    />
    <ol
      class="breadcrumb breadcrumbs-root"
    >
      <li
        class="breadcrumb-item"
      >
        <a
          class="breadcrumb-text-truncate breadcrumb-link"
          href="/workspace/23/settings/definitions"
        >
          Definitions
        </a>
      </li>
      <li
        class="breadcrumb-item active"
      >
        <span
          class="breadcrumb-text-truncate"
        >
          Events
        </span>
      </li>
    </ol>
    <div
      class="content-header"
    >
      <div
        class="header-text"
      >
        <div
          class="d-flex"
        >
          <h3
            class="title-text text-truncate"
          >
            <span
              class="text-truncate"
              data-tooltip="false"
              title=""
            >
              Events
            </span>
          </h3>
        </div>
        <div
          class="description"
        >
          This is the data model of events sent to Analytics Cloud.
        </div>
      </div>
      <div
        class="page-actions-container"
      />
    </div>
    <div>
      <div
        class="card card-root page-display"
      >
        <div
          class="d-flex justify-content-between"
        >
          <ul
            class="nav-root page-subnav mx-4 my-3 nav nav-underline"
            role="tablist"
          >
            <li
              class="nav-item"
              role="presentation"
            >
              <a
                class="nav-link"
                href="/workspace/23/settings/definitions/events/default"
                role="tab"
              >
                <div
                  class="mb-2"
                >
                  <b>
                    Default Events
                  </b>
                </div>
              </a>
            </li>
            <li
              class="nav-item"
              role="presentation"
            >
              <a
                class="nav-link"
                href="/workspace/23/settings/definitions/events/custom"
                role="tab"
              >
                <div
                  class="mb-2"
                >
                  <b>
                    Custom Events
                  </b>
                </div>
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
